x/emissions/keeper/inference_synthesis/network_inferences.go

```diff
@@ -1,200 +1,243 @@
 package inferencesynthesis
 
 import (
-	"errors"
 	"fmt"
 
 	"cosmossdk.io/collections"
 	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/pkg/errors"
+
 	alloraMath "github.com/allora-network/allora-chain/math"
+	"github.com/allora-network/allora-chain/utils/fn"
 	emissionskeeper "github.com/allora-network/allora-chain/x/emissions/keeper"
 	emissions "github.com/allora-network/allora-chain/x/emissions/types"
-	sdk "github.com/cosmos/cosmos-sdk/types"
-	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
+type GetNetworkInferencesResult struct {
+	NetworkInferences                    *emissions.ValueBundle
+	ForecasterToForecastImpliedInference map[string]*emissions.Inference
+	InfererToWeight                      map[Inferer]Weight
+	ForecasterToWeight                   map[Forecaster]Weight
+	InferenceBlockHeight                 int64
+	LossBlockHeight                      int64
+}
+
 func GetNetworkInferences(
 	ctx sdk.Context,
 	k emissionskeeper.Keeper,
 	topicId TopicId,
 	inferencesNonce *BlockHeight,
-) (
-	networkInferences *emissions.ValueBundle,
-	forecasterToForecastImpliedInference map[string]*emissions.Inference,
-	infererToWeight map[Inferer]Weight,
-	forecasterToWeight map[Forecaster]Weight,
-	inferenceBlockHeight int64,
-	lossBlockHeight int64,
-	err error,
-) {
-	// Decide whether to use the latest inferences or inferences at a specific block height
-	var inferences *emissions.Inferences
+) (*GetNetworkInferencesResult, error) {
+	// Retrieve the requested inferences (either latest or specified, depending on inferencesNonce)
+	inferences, inferenceBlockHeight, err := getRequestedInferences(ctx, k, topicId, inferencesNonce)
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "while getting inferences")
+	}
+
+	if len(inferences.Inferences) > 1 {
+		// If we have multiple inferences:
+		// 1. Try to get latest network loss
+		networkLosses, err := k.GetLatestNetworkLossBundle(ctx, topicId) // TODO(spook): why latest?  why not use `inferenceBlockHeight`?
```

Comment:
Would simply have to cache the network loss bundle in a new KV by nonce at CloseReputerRequest time, which we can do. Would be more efficient that way

Reply:
Actually, it was something from research. It is allowed to use the latest instead of just the exact past one with the exact nonce.

Reply:
Actually, it was something from research. It is allowed to use the latest instead of just the exact past one.